### PR TITLE
Add atomic-deployment flag

### PR DESCRIPTION
**Pull request by tobiscr**
**Description**
Add support for atomic deployments.

**Related issue(s)**
https://github.com/kyma-project/cli/issues/755

**Depends on**
https://github.com/kyma-incubator/hydroform/pull/198

### CODE REVIEW

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 1]
```diff
@@ -15,6 +15,7 @@ kyma alpha deploy [flags]
 ## Options
 
 ```bash
+  -a, --atomic                    Use atomic deployment (rollback any component which could not be successfully installed). (default true)
```
Comment (NHingerl) [inline docs/gen-docs/kyma_alpha_deploy.md:18]:
```suggestion
  -a, --atomic                    Use atomic deployment, which rolls back any component that could not be installed successfully (default: true).
```

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ kyma alpha deploy [flags]
...
       --cancel-timeout duration   Time after which the workers' context is canceled. Pending worker goroutines (if any) may continue if blocked by a Helm client. (default 15m0s)
```
Comment (NHingerl) [inline docs/gen-docs/kyma_alpha_deploy.md:19]: Does the timeout affect several workers or just one worker? If just one, use "worker's" instead of "worker'"

```suggestion
      --cancel-timeout duration   Time after which the workers' context is canceled (default: 15m0s). Any pending worker goroutines that are blocked by a Helm client will continue.
```

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 3]
```diff
@@ -15,6 +15,7 @@ kyma alpha deploy [flags]
...
   -c, --components string         Path to the components file. (default "workspace/installation/resources/components.yaml")
```
Comment (NHingerl) [inline docs/gen-docs/kyma_alpha_deploy.md:20]:
```suggestion
  -c, --components string         Path to the components file (default: "workspace/installation/resources/components.yaml")
```

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 4]
```diff
@@ -15,6 +15,7 @@ kyma alpha deploy [flags]
...
   -d, --domain string             Domain used for installation. (default "local.kyma.dev")
```
Comment (NHingerl) [inline docs/gen-docs/kyma_alpha_deploy.md:21]:
```suggestion
  -d, --domain string             Domain used for installation (default: "local.kyma.dev")
```

**Review by NHingerl — CHANGES_REQUESTED** [review]
Not quite sure about my 2nd sentence for the timeout, pls check if I got it right.

File: docs/gen-docs/kyma_alpha_deploy.md (hunk 2 above)
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_deploy.md:19]: the timeout affects all workers

File: docs/gen-docs/kyma_alpha_deploy.md (hunk 3 above)
Comment (tobiscr) [reply docs/gen-docs/kyma_alpha_deploy.md:20]: Good input, but we need to handled that decoupled form this PR as it requires a change in the doc-generator logic.

**Comment by tobiscr** [comment]
**TL;DR**
This PR will be discontinued and is now part of #763 .

**Long version**
We had a major change in the Hydroform API (logging was changed to use ZAP) which requires a bigger change on CLI side. To avoid duplicated work, I will close this PR and merge its changes into PR #763. This PR includes all changes to integrate the CLI with the latest Hydroform API.

@NHingerl : thanks for your valuable feedback - is it also added to PR #763. Please review the updates there.
